python/cudf/cudf/core/dataframe.py

```diff
@@ -526,11 +525,13 @@ def serialize(self):
 
         # Use the column directly to avoid duplicating the index
         # need to pickle column names to handle numpy integer columns
-        header["column_names"] = pickle.dumps(tuple(self._data.names))
-        column_header, column_frames = column.serialize_columns(self._columns)
-        header["columns"] = column_header
+        header["columns"], column_frames = column.serialize_columns(
+            self._columns
+        )
+        header["column_frame_count"] = len(column_frames)
         frames.extend(column_frames)
 
+        header["column_names"] = pickle.dumps(tuple(self._data.names))
```

Comment:
Are these changes meant to be meaningful, aside from adding the count to the header (not sure why)?

Reply:
I think it resulted from an (not-successful) attempt to refactor the serialization methods. Realigning the lines to make them look more alike to the series counterparts. Removing the column counts in this case.